plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RewriteCast.java

```diff
@@ -66,6 +74,11 @@ private boolean pushdownSupported(JdbcTypeHandle sourceType, Type targetType)
         return switch (targetType) {
             case SmallintType _, IntegerType _, BigintType _ ->
                     SUPPORTED_SOURCE_TYPE_FOR_INTEGRAL_CAST.contains(sourceType.jdbcType());
+            // varchar -> char is unsupported as varchar supports multi-byte characters whereas char supports only single byte characters.
+            case CharType _ -> CHAR == sourceType.jdbcType();
+            // char -> varchar is not supported as Redshift doesn't pad char value with blanks whereas Trino pads char value with blanks.
+            // cast to unbounded varchar is unsupported as Redshift doesn't support unbounded varchar
+            case VarcharType varcharType -> VARCHAR == sourceType.jdbcType() && !varcharType.isUnbounded();
```

Comment:
And in case of bounded varchar, shouldn't we restrict them to the maximum length redshift supports ?

Reply:
Pushdown for cast to unbounded varchar is not supported in this PR. We can handle it when we add support. Please advise if we still need to map unbounded to max supported precision.

Reply:
I think we can support coercion till the target varchar length is less the max supported precision - post that we could restrict the pushdown. WDYT ?

Reply:
that makes sense. I revised a condition to restrict upto `REDSHIFT_MAX_VARCHAR` varchar length.